packages/ng/numeric-badge/numeric-badge.component.ts

```diff
@@ -27,4 +29,8 @@ export class NumericBadgeComponent {
 	 * The palette to use for this badge. Defaults to 'none' (inherits parent palette)
 	 */
 	palette: Palette = 'none';
+
+	ngOnChanges(): void {
+		this.#ngClass.ngClass = ['numericBadge', `palette-${this.palette}`, `mod-${this.size}`];
```

Comment:
We can't, that's the pain point.

Because `#ngClass.ngClass` is just a setter and you cannot edit the state of the class without overwriting it, meaning that we have to compute it everytime for the directive to process it entirely everytime.

See https://github.com/angular/angular/blob/16.2.9/packages/common/src/directives/ng_class.ts#L81

Reply:
Hmm ok could we add least do something like this to avoid updating it when other Input properties (like value) change ? 
```suggestion
	ngOnChanges(changes: SimpleChanges): void {
		if (changes.palette.currentValue || changes.size.currentValue) {
			const newClasses = ['numericBadge'];
			if (changes.palette.currentValue) {
				newClasses.push(`palette-${changes.palette.currentValue}`);
			}
			if (changes.size.currentValue) {
				newClasses.push(`mod-${changes.size.currentValue}`);
			}
			this.#ngClass.ngClass = newClasses
		}
	}
```
(top of my head, could be written better)

Reply:
We can't, because if only `size` changed, then `ngClass` becomes `['numericBadge', 'mod-XS']` for instance, which is missing the palette.

And since we're sending a new array everytime (same for the object approach), we have to compute it all over again. The goal being to do our best to not make expensive computing in here.

Reply:
Ah yes you're right, forgot about this edge case (where if no changes are there, we should use `this.palette` or `this.size`, my bad. This should work :

```suggestion
	ngOnChanges(changes: SimpleChanges): void {
		if (changes.palette.currentValue || changes.size.currentValue) {
			this.#ngClass.ngClass = ['numericBadge', `palette-${this.palette}`, `mod-${this.size}`];
		}
	}
```